Scripts/UtilityScripts/AzTSUtility.ps1

```diff
@@ -0,0 +1,1617 @@
+    .EXAMPLE
+    PS> Get-DataFactoryV2DataSets -SubscriptionId "00000000-xxxx-0000-xxxx-000000000000" -ResourceGroupName "MyResourceGroupName" -DataFactoryName "MyDataFactoryName"
+    .LINK
+    None
+  #>
+
+  [CmdletBinding()]
+  param (
+      [Parameter(Mandatory=$true)]
+      [string]
+      $SubscriptionId,
+      [Parameter(Mandatory=$true)]
+      [string]
+      $ResourceGroupName,
+      [Parameter(Mandatory=$true)]
+      [string]
+      $DataFactoryName
+  )
+
+  $profile = Set-AzContext -Subscription $SubscriptionId
+
+  $datasets = Get-AzDataFactoryV2Dataset -ResourceGroupName $ResourceGroupName -DataFactoryName $DataFactoryName
+
+  foreach ($dataset in $datasets)
+  {
+    Write-Information -InformationAction Continue -MessageData ("Dataset: " + $dataset.Name)
+    Write-Information -InformationAction Continue -MessageData "Parameter Names and Values:"
+    foreach ( $h in $dataset.Properties.Parameters.GetEnumerator() )
+    {
+      Write-Information -InformationAction Continue -MessageData "- $($h.Key) = $($h.Value.DefaultValue)"
+    }
+    Write-Information -InformationAction Continue -MessageData ""
+  }
+}
+
+function Get-DataFactoryV2LinkedServices()
+{
+  <#
+    .SYNOPSIS
+    This command lists linked service names and parameters for linked services that have at least one parameter.
+    .DESCRIPTION
+    This command lists linked service names and parameters for linked services that have at least one parameter. This is to review parameter values to try and locate parameters that may contain strings identified by the control as plaintext secrets.
+    .PARAMETER SubscriptionId
+    The Azure subscription ID containing the Data Factory.
+    .PARAMETER ResourceGroupName
+    The Resource Group name containing the Data Factory.
+    .PARAMETER DataFactoryName
+    The Data Factory name.
+    .INPUTS
+    None
+    .OUTPUTS
+    For each Linked Service, text with the Linked Service name and its parameter names and values.
+    .EXAMPLE
+    PS> Get-DataFactoryV2LinkedServices -SubscriptionId "00000000-xxxx-0000-xxxx-000000000000" -ResourceGroupName "MyResourceGroupName" -DataFactoryName "MyDataFactoryName"
+    .LINK
+    None
+  #>
+
+  [CmdletBinding()]
+  param (
+      [Parameter(Mandatory=$true)]
+      [string]
+      $SubscriptionId,
+      [Parameter(Mandatory=$true)]
+      [string]
+      $ResourceGroupName,
+      [Parameter(Mandatory=$true)]
+      [string]
+      $DataFactoryName
+  )
+
+  $profile = Set-AzContext -Subscription $SubscriptionId
+
+  $linkedServices = Get-AzDataFactoryV2LinkedService -ResourceGroupName $ResourceGroupName -DataFactoryName $DataFactoryName
+
+  foreach ($linkedService in $linkedServices)
+  {
+    Write-Information -InformationAction Continue -MessageData ("Linked Service: " + $linkedService.Name)
+    Write-Information -InformationAction Continue -MessageData "Parameter Names and Values:"
+    foreach ( $h in $linkedService.Properties.Parameters.GetEnumerator() )
```

Comment:
Null check might be required, same as above.

Reply:
Thanks - mitigated